src/main/java/pl/tomaszdziurko/jvm_bloggers/blog_posts/AggregatedRssFeedProducer.java

```diff
@@ -64,31 +70,40 @@
      * @param limit Upper limit of RSS entries count in the generated feed. If equal or less
      *     than zero then all approved blog posts will be generated.
      *
+     * @param excludedAuthors RSS entries for given authors will be excluded
+     *     from the generated feed
+     *
      * @return Aggregated RSS feed for approved blog posts ordered by publication date
      */
     @Cacheable
-    public SyndFeed getRss(String feedUrl, int limit) {
+    public SyndFeed getRss(String feedUrl, int limit, Set<String> excludedAuthors) {
 
         Preconditions.checkArgument(
             StringUtils.isNotBlank(feedUrl), "feedUrl parameter cannot be blank");
 
-        final StopWatch stopWatch = new StopWatch();
-        log.debug("Building aggregated RSS feed...");
-        stopWatch.start();
+        StopWatch stopWatch = null;
+        if (log.isDebugEnabled()) {
+            stopWatch = new StopWatch();
+            log.debug("Building aggregated RSS feed...");
+            stopWatch.start();
+        }
 
         final Pageable pageRequest = new PageRequest(0, limit > 0 ? limit : Integer.MAX_VALUE);
         final List<BlogPost> approvedPosts =
-            blogPostRepository.findByApprovedTrueOrderByPublishedDateDesc(pageRequest);
+            blogPostRepository.findByApprovedTrueAndBlogAuthorNotInOrderByPublishedDateDesc(
+                pageRequest, firstNonNull(excludedAuthors, EMPTY_AUTHORS_SET)
```

Comment:
In my opinion the controller here acts only as a pass-through delegate. It passes the parameters as is. Following your approach we should also move `limit` parameter checking to the controller to be consistent with your approach. Additionally there is some magic related to the underlying database here which will be exposed out to the controller unnecessarily. Take a look at [`EMPTY_AUTHORS_SET`](https://github.com/tdziurko/jvm-bloggers/pull/215/files#diff-bb1b427605a1cafe46d0c0b54e6c5125R58) initialization. It is a special set with "" (empty string) element required to properly generate SQL statement. I don't want to expose it outside this class.